custom_components/octopus_intelligent/config_flow.py

```diff
@@ -58,13 +59,13 @@ async def async_step_user(self, user_input=None):
 
         errors = {}
         try:
-            await self.hass.async_add_executor_job(
-                try_connection,
-                user_input[CONF_API_KEY],
-                user_input[CONF_ACCOUNT_ID])
-        except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception("Unexpected exception")
-            errors["base"] = "unknown"
+            await try_connection(user_input[CONF_API_KEY], user_input[CONF_ACCOUNT_ID])
+        except Exception as ex:  # pylint: disable=broad-except
+            _LOGGER.error(ex)
+            if isinstance(ex, InvalidAuthError):
+                errors["base"] = "invalid_auth"
+            else:
+                errors["base"] = "unknown"
```

Comment:
The removal of `async_add_executor_job` fixes the following _"coroutine 'try_connection' was never awaited”_ error in the HASS logs:
```
2024-02-27 22:36:00.858 WARNING (MainThread) [py.warnings] /usr/local/lib/python3.12/asyncio/base_events.py:1975: RuntimeWarning: coroutine 'try_connection' was never awaited
  handle = None  # Needed to break cycles when an exception occurs.
```
The `try_connection()` function was simply not being called. `async_add_executor_job()` creates a new thread to run the function, but it does not expect the function to be asynchronous and does not await it. When not awaited, asynchronous functions like `try_connection()` are not executed.

The introduction of `InvalidAuthError` discriminates between “expected” authentication errors and unexpected / unknown errors.